implementation/src/main/java/com/teampc/controller/ViewSubmissionsController.java

```diff
@@ -17,27 +20,68 @@
 
    private Test currentTest;
 
+   @FXML
+   private Text testName;
+
+   @FXML
+   private Text className;
+
+   private String getTestDisplayStr(Test t) {
+      if (t == null) {
+         return "null Test";
+      }
+
+      String str = "";
+      if (t.getName() != null) {
+         str += t.getName();
+      }
+      if (t.getCourse() != null) {
+         str += " - " + t.getCourse();
+      }
+      return str;
+   }
+
    @Override
    public void initialize(URL location, ResourceBundle resources) {
       data = FXCollections.observableArrayList();
    }
 
+   // canned data
    private Submission s;
 
    // todo: alright, let's fix this later :(
    public void setCurrentTest(Test test) {
       this.currentTest = test;
 
-      this.s = new Submission();
-      this.s.responses.add(new MultipleChoiceQuestionResponse());
-      this.s.responses.add(new MultipleChoiceQuestionResponse());
-      this.s.taker = UserSession.loggedInUser;
+      testName.setText(getTestDisplayStr(currentTest));
+      className.setText(currentTest.getCourseName());
+
+      Collection<Submission> submissions = SubmissionDAO.getInstance().fetchSubmissionsForTest(currentTest);
```

Comment:
> I don't care if we also show the key in the submissions. If it really bugs you, we can consider fixing this.
>  Add a line note

doesn't bug me, just wan't sure if it was intentional or would break anything